File: src/main/java/io/github/foodjournal/application/JournalIntent.java
package io.github.foodjournal.application;
import java.time.Instant; import java.util.List;
public record JournalIntent(IntentType type, String mealDescription, Integer calories, String query, Long entryId, String timezone, Integer calorieTarget, Boolean reportsEnabled, List<MealItem> items, Instant eatenAt, String nutritionSource, String confidence) {
 public JournalIntent(IntentType type,String mealDescription,Integer calories,String query,Long entryId,String timezone,Integer calorieTarget,Boolean reportsEnabled,List<MealItem> items){this(type,mealDescription,calories,query,entryId,timezone,calorieTarget,reportsEnabled,items,null,null,null);}
 public record MealItem(String name, Double grams, Integer calories, Double proteinGrams, Double carbsGrams, Double fatGrams) { public MealItem(String name,Double grams,Integer calories){this(name,grams,calories,null,null,null);} }
}

File: python-app/app/domain/journal_intent.py
from dataclasses import dataclass


@dataclass(frozen=True)
class MealItem:
    name: str | None
    grams: float | None
    total_calories: int | None = None
    calories_per_100g: int | None = None
    protein_grams: float | None = None
    carbs_grams: float | None = None
    fat_grams: float | None = None
    barcode: str | None = None
